main.go

```diff
@@ -413,6 +450,51 @@ func convert(req *plugin.CodeGeneratorRequest) (*plugin.CodeGeneratorResponse, e
 	return res, nil
 }
 
+func addExtensions(msg *descriptor.DescriptorProto, extensions []*descriptor.FieldDescriptorProto, pkg string) {
+	for _, ext := range extensions {
+		extendeeNodes := strings.Split(ext.GetExtendee(), ".")
+		pkgNodes := strings.Split(pkg, ".")
+
+		// We already know this extendee cannot be in this package.
+		if len(pkgNodes) > len(extendeeNodes) {
+			continue
+		}
+
+		// skipping leading "."
+		nodeIdx := 1
+		for _, node := range pkgNodes {
+			if node != extendeeNodes[nodeIdx] {
+				nodeIdx = 0
+				break
+			}
+			nodeIdx += 1
+		}
+
+		// Not the right package or message
+		if nodeIdx == 0 || msg.GetName() != extendeeNodes[nodeIdx] {
+			continue
+		}
+
+		foundNode := (*descriptor.DescriptorProto)(nil)
+		nestedMsgs := msg.GetNestedType()
+		for i := nodeIdx; i < len(extendeeNodes); i += 1 {
+			foundNode = nil
```

Comment:
you seem to reset it to nil and the break below only breaks the inner for loop, is this expected?

Reply:
Yes, it is expected behavior